src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -286,23 +299,33 @@ def _rbd_image(cls, ioctx, pool_name, namespace, image_name):
             # snapshots
             stat['snapshots'] = []
             for snap in img.list_snaps():
+                if img.mirror_image_get_mode() == rbd.RBD_MIRROR_IMAGE_MODE_JOURNAL:
+                    snap['mirror_mode'] = 'journal'
+                elif snapshot_mode:
+                    snap['mirror_mode'] = 'snapshot'
+                else:
+                    snap['mirror_mode'] = 'unknown'
```

Comment:
Let's reuse the enum and let's not invent a new value ('unknown' is not a valid RBD mode, so it's better to loudly shout an exception than to silently -try to- deal with errors, aka fail fast approach):
```suggestion
                try:
                    snap['mirror_mode'] = MIRROR_IMAGE_MODE(mirror_mode)
                except ValueError as ex:
                    raise DashboardError(f"Unknown RBD Mirror mode: {ex}") 
```